graphql-mode.el

```diff
@@ -51,22 +51,25 @@
 
 (defcustom graphql-indent-level 2
   "Number of spaces for each indentation step in `graphql-mode'."
+  :tag "GraphQL"
   :type 'integer
   :safe 'integerp)
 
+(defcustom graphql-url "http://localhost:8000/graphql"
+  "URL address of the graphql server endpoint."
+  :tag "GraphQL"
+  :type 'string)
 
-(defvar graphql-url
-  nil)
-(make-variable-buffer-local 'graphql-url)
-
-(defun graphql--query (query)
+(defun graphql--query (query operation variables)
   "Send QUERY to the server at `graphql-url' and return the
 response from the server."
   (let* ((url-request-method "POST")
-	 (query (url-encode-url query))
-	 (operation (graphql-current-operation))
-	 (variables (url-encode-url (graphql-current-variables)))
-	 (url (format "%s?query=%s&operationName=%s&variables=%s" graphql-url query operation variables)))
+         (query (url-encode-url query))
+         (url (format "%s?query=%s" graphql-url query)))
+    (if operation
+        (setq url (concat url "&operationName=" operation)))
+    (if variables
+        (setq url (concat url "&variables=" (url-encode-url variables))))
     (with-current-buffer (url-retrieve-synchronously url t)
```

Comment:
by default is the anonymous  query,
then if there is named queries, look for the current query, where the cursor is between the pair of "{" "}".
then should ask for the named query (TBD)